calico_containers/pycalico/netns.py

```diff
@@ -61,191 +60,153 @@ def setup_logging(logfile, level=logging.INFO):
     _log.addHandler(handler)
 
 
-def remove_endpoint(ep_id):
+def create_veth(veth_name_host, veth_name_ns_temp):
     """
-    Remove an endpoint.
+    Create the veth (pair).
+    :param veth_name_host: The name of the veth interface
+    :param veth_name_ns_temp: The temporary interface name of the veth that will be
+    moved into the namespace.
+    :return: None. Raises CalledProcessError on error.
+    """
+    # Create the veth
+    check_call(['ip', 'link',
+                'add', veth_name_host,
+                'type', 'veth',
+                'peer', 'name', veth_name_ns_temp],
+               timeout=IP_CMD_TIMEOUT)
+
+    # Set the host end of the veth to 'up' so felix notices it.
+    check_call(['ip', 'link', 'set', veth_name_host, 'up'],
+               timeout=IP_CMD_TIMEOUT)
 
-    :param ep_id: The endpoint ID to remove
-    :return: Nothing
+
+def remove_veth(veth_name_host):
+    """
+    Remove the veth (pair).
+    :param interface_name: The name of the veth interface.
+    :return: None. Raises CalledProcessError on error.
     """
-    iface = IF_PREFIX + ep_id[:11]
-    call("ip link delete %s" % iface, shell=True)
+    # The veth removal is best effort. If it fails then just log.
+    check_call(['ip', 'link', 'del', veth_name_host], timeout=IP_CMD_TIMEOUT)
 
 
-def add_ip_to_interface(container_pid, ip, interface_name,
-                        proc_alias=PROC_ALIAS):
+#TODO Do we need to make proc_alias configurable anymore?
+def move_veth_into_ns(cpid,
+                      veth_name_ns_temp,
+                      veth_name_ns,
+                      proc_alias=PROC_ALIAS):
     """
-    Add an IP to an interface in a container.
+    Move the veth into the namespace.
 
-    :param container_pid: The PID of the namespace to operate in.
-    :param ip: The IPAddress to add.
-    :param interface_name: The interface to add the address to.
+    :param cpid: The PID of a process currently running in the namespace.
+    :param veth_name_ns_temp: The temporary interface name of the veth that will be
+    moved into the namespace.
+    :param veth_name_ns: The name of the interface in the namespace.
     :param proc_alias: The location of the /proc filesystem on the host.
-    :return: None. raises CalledProcessError on error.
+    :return: None. Raises CalledProcessError on error.
     """
-    with NamedNamespace(container_pid, proc=proc_alias) as ns:
-        ns.check_call("ip -%(version)s addr add %(addr)s/%(len)s "
-                      "dev %(device)s" %
-                        {"version": ip.version,
-                         "len": PREFIX_LEN[ip.version],
-                         "addr": ip,
-                         "device": interface_name},
-                      shell=True)
+    # Configure the container end.  To do this we create a temporary link from
```

Comment:
As per the TODO and comment above, does that mean we can get rid of this NamedNamespace thing?